### PR TITLE
PHOENIX-2257 Fix broken integration test for query over derived GROUP…

### PR DESCRIPTION
**Pull request by chiastic-security**
… BY table (testDerivedTableWithGroupBy()), which worked on Java 7 but not on Java 8.

The test was retrieving two rows from a query with no ORDER BY clause, and assuming that they would come back in a specific order. This happened to match the order that Java 7 returned them in, but the order has changed in Java 8.

Test fixed by not assuming any particular order.

### CODE REVIEW

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/DerivedTableIT.java [hunk 1]
```diff
@@ -54,14 +28,17 @@
 import org.junit.runners.Parameterized;
 import org.junit.runners.Parameterized.Parameters;
 
-import com.google.common.collect.Lists;
+import java.sql.*;
+import java.util.*;
```
Comment (JamesRTaylor) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/DerivedTableIT.java]: Please see https://phoenix.apache.org/contributing.html#Code_conventions. You'll want to conform to these - easiest way is to import dev/eclipse_prefs_phoenix.epf and dev/phoenix.importorder into Eclipse. This is basically to ensure we don't get a lot of code differences due to different import ordering and whitespacing.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/DerivedTableIT.java [hunk 2]
```diff
@@ -313,21 +290,25 @@ public void testDerivedTableWithGroupBy() throws Exception {
             query = "SELECT DISTINCT COLLECTDISTINCT(t.b) FROM (SELECT b_string b, a_string a FROM aTable GROUP BY a_string, b_string) AS t GROUP BY t.a";
             statement = conn.prepareStatement(query);
             rs = statement.executeQuery();
-            assertTrue (rs.next());
+            Set<Array> queryResults = new HashSet<>();
```
Comment (JamesRTaylor) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/DerivedTableIT.java]: Use BaseTest.assertValuesEqualsResultSet() instead which does the same thing as this.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/DerivedTableIT.java [hunk 3]
```diff
@@ -313,21 +290,25 @@ public void testDerivedTableWithGroupBy() throws Exception {
             query = "SELECT DISTINCT COLLECTDISTINCT(t.b) FROM (SELECT b_string b, a_string a FROM aTable GROUP BY a_string, b_string) AS t GROUP BY t.a";
```
Comment (JamesRTaylor) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/DerivedTableIT.java]: @maryannxue - do you think the underlying order returned is indeterminate? For our tests, we have phoenix.query.force.rowkeyorder set to true, so underlying scans will return in row key order. Just want to make sure @chiastic-security fix isn't masking a bug.

Comment (maryannxue) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/DerivedTableIT.java]: @JamesRTaylor Do we have a deterministic ordering for arrays? If we do, I don't think this change should be made without finding out the cause.

Comment (maryannxue) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/DerivedTableIT.java]: @JamesRTaylor Please ignore my previous (deleted) comment. I just realized that the server group-by has replaced tree map with hash map, which means an optimization I did for client group-by based on the assumption that the input is a sorted group-by is now invalid. I will fix this ASAP. And after the fix, the test case should fine, we should keep it as it is.

**Comment by stoty** [comment]
Already merged.

**Pull request closed without merging**